modules/apps/fragment/fragment-collection-contributor/.eslintrc.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+module.exports = {
+	globals: {
+		YT: true,
+		configuration: true,
+		fragmentElement: true,
+		fragmentNamespace: true,
+	},
+	overrides: [
+		{
+			files: '**/src/**/*.js',
+			rules: {
+
+				// For IE compatibility because JS here doesn't get transpiled.
+
+				'notice/notice': 'off',
+				'object-shorthand': 'off',
+				'prefer-arrow-callback': 'off',
+				'prefer-object-spread': 'off',
```

Comment:
@jkappler: These are the overrides I was telling you about for everything under `/fragment-collection-contributor`. The comment there:

```
// For IE compatibility because JS here doesn't get transpiled.
```

is not 100% accurate, because the first rule is actually for suppressing the licensing headers in the snippets inserted into the Fragment Editor. The other three rules are about stopping ESLint from making changes that would break IE:

- `{a: a}` → `{a}`
- `function() {}` → `() => {}`
- `Object.assign({}, obj)` -> `{...obj}`

Reply:
cc @victorg1991 — I don't want to spoil any of the good work you did in c3129efdb0ade32eec14cf06b.

And cc @sandrodw3 too at the recommendation of @jkappler.

Reply:
Is this going to get backported? 

This is problematic because the implication is that:
a) We can't forget about IE11 in 7.4
b) Every backport to 7.3 from `master` is very risky and not properly audited... lots of IE11 issues can get through

Reply:
It could be backported, but _this_ change makes us safer (for now) because previously, there was nothing stopping somebody accidentally sticking an arrow function in here.

When we remove IE support from master (soon), _that_ change will be able to remove these rules, and _that_ change obviously must not be backported.

Reply:
(That's actually one of the reasons why I put the "IE" comment in here, so it will turn up when we grep for `\bIE(6|7|8|9|10|11)?\b` in the future).

Reply:
That's what I mean... it makes us safer at the cost of relaxing some of the rules we enforce everywhere else, just in the off-chance this gets backported to `7.3`

Reply:
> (That's actually one of the reasons why I put the "IE" comment in here, so it will turn up when we grep for \bIE(6|7|8|9|10|11)?\b in the future).

Based on the reasoning, that would be 7 years in the future, which is how long `7.3` can last... 

Maybe a plausible approach would be to migrate this config to `7.3.x` and keep `master` clean?

Reply:
I don't get that reasoning at all. I'm talking about the "Remove IE compatibility" epic (no link because I'm on my phone) within days or weeks, not years.

Master is not clean now and it won't be after either (we still need this `.eslintrc.js` after removing IE support because we have to keep the override of the "notice/notice" rule.

I just dont think anything has changed with respect to IE yet as a result of 7.3 release being cut; it changes when we do the related epic, and at that point backporting behavior will have to change across the board...

(FWIW if you look at the LPS, Brian wants the coverage increased on older branches too, so it's probably going to be me doing the backporting of this specific change.)

Reply:
> it makes us safer at the cost of relaxing some of the rules we enforce everywhere else

Note that we enforced no rules at all here before, so it's not exactly a relaxation — we are strictly increasing the enforcement here.

Reply:
> I don't get that reasoning at all. I'm talking about the "Remove IE compatibility" epic (no link because I'm on my phone) within days or weeks, not years.

We're on the same page, then. The usage of the word "future" made me think about how would we prevent IE-unsafe code from being backported into 7.3.

Reply:
As in, it's unsafe in `7.3.x` but totally safe in `master`...